### DISCUSSION TITLE
Nuclei is not getting updated

### DISCUSSION THREAD

**Discussion by HunterX461** [projectdiscovery/nuclei#5274] [Q&A]
### Current Behavior: while trying to update the nuclei  to its latest v3.2.8 it is not updating and showing an error message 'flag provided but not defined: -update'

The nuclei -version is giving this as output  :

[INF] Nuclei Engine Version: v3.2.7
[INF] Nuclei Config Directory: /root/.config/nuclei
[INF] Nuclei Cache Directory: /root/.cache/nuclei
[INF] PDCP Directory: /root/.pdcp

while the nuclei -ut is just showing this message and nothing else :

Your current nuclei-templates  are outdated. Latest is v9.8.8
[ERR] Could not find template '/home/x/.local/nuclei-templates': no templates found in path /home/x/.local/nuclei-templates
[INF] Current nuclei version: v3.2.7 (outdated)
[INF] Current nuclei-templates version:  (outdated)
[WRN] Scan results upload to cloud is disabled.
[INF] No results found. Better luck next time!
[FTL] Could not run nuclei: no templates provided for scan

The above stated output is also shown for 
nuclei templates reset  and nuclei -ud  /.local/nuclei-templates

### Expected Behavior:

After using the nuclei -update command ,the nuclei engine must update to its latest version also same thing must happen with nuclei -ut and the templates must update to their latest version

### Steps To Reproduce:

make sure you have nuclei v3.2.7
run nuclei -ut or nuclei -update 

### Anything else:

![Screenshot 2024-06-08 125258](https://github.com/projectdiscovery/nuclei/assets/105849365/bb05d9f9-b275-41d4-a1bf-e8da35589ef3)
![Screenshot 2024-06-08 125347](https://github.com/projectdiscovery/nuclei/assets/105849365/20958c9b-6ba4-4b14-b254-8fcc53e2392b)
![Screenshot 2024-06-08 125035](https://github.com/projectdiscovery/nuclei/assets/105849365/69d67c4f-466e-46a4-b595-851b86d5b47b)

**Comment by ehsandeep** [comment]
See https://github.com/projectdiscovery/nuclei/discussions/2151, update option is patched in kali, so you need to install nuclei using another method like via [pdtm](https://github.com/projectdiscovery/pdtm) or `go install` to avoid this.

**Reply by HunterX461** [reply]
i tried to do that using pdtm but still the older nuclei version is visible and the templates are also the older ones

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
